### PR TITLE
feat: show resource utilization in DevTools plugin

### PR DESCRIPTION
**Pull request by drodil**
## Hey, I just made a Pull Request!

Shows resource utilization on devtools page.

image

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by freben** [comment]
With the caveat that it'll of course show the info for only the one host out of potentially many, that the request happened to go to.

**Comment by drodil** [comment]
@freben yep. But the same applies to all the other info like OS, config, and dependencies.

**Review by awanlin — APPROVED** [review]
This is a nice little addition, thanks @drodil 🚀

**Comment by awanlin** [comment]
FYI - The E2E failures are related to this: https://github.com/remix-run/react-router/issues/10566#issuecomment-1593277043

**Comment by freben** [comment]
Waiting for getting back from vacations 😅

**Comment by drodil** [comment]
Same here, it's vacation time! 😎

**Comment by efenner-cambia** [comment]
Uh oh. @drodil did you mean to let this get closed?

**Comment by drodil** [comment]
@efenner-cambia no. I've been on vacation so my activity has been zero for the past month 😎 Could someone re-open?

**Comment by drodil** [comment]
Remove stale, waiting for review.

File: plugins/devtools-backend/src/api/DevToolsBackendApi.ts [hunk 1]
```diff
@@ -204,7 +204,16 @@ export class DevToolsBackendApi {
...
+    const operatingSystem = `${os.hostname()}: ${os.type} ${os.release} - ${
+      os.platform
+    }/${os.arch}`;
+    const usedMem = Math.floor((os.totalmem() - os.freemem()) / (1024 * 1024));
+    const resources = `Memory: ${usedMem}/${Math.floor(
+      os.totalmem() / (1024 * 1024),
+    )}MB - Load: ${os
+      .loadavg()
+      .map(v => v.toFixed(2))
+      .join('/')}`;
```
[the remark this answers is not in the document]
Comment (awanlin) [reply plugins/devtools-backend/src/api/DevToolsBackendApi.ts:216]: @drodil any thoughts on this?

**Comment by awanlin** [comment]
@freben @Rugvip with @jhaals being away on vacation can you give this a review please?

File: plugins/devtools-backend/src/api/DevToolsBackendApi.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (drodil) [reply plugins/devtools-backend/src/api/DevToolsBackendApi.ts:216]: Not really, I think most instances run on machines that have gigabytes of memory instead some smaller amount. And overall, it's just a snapshot of the current resource usage so it's just giving some idea of what is going on thus don't think this is major issue.

**Comment by drodil** [comment]
Nope, still not stale.

File: plugins/devtools-backend/src/api/DevToolsBackendApi.ts [hunk 2]
```diff
@@ -204,7 +204,16 @@ export class DevToolsBackendApi {
   }
 
   public async listInfo(): Promise<DevToolsInfo> {
-    const operatingSystem = `${os.type} ${os.release} - ${os.platform}/${os.arch}`;
+    const operatingSystem = `${os.hostname()}: ${os.type} ${os.release} - ${
```
[the remark this answers is not in the document]
Comment (jhaals) [reply plugins/devtools-backend/src/api/DevToolsBackendApi.ts:207]: Not an overly strong opinion about this but aren't operating system something that you normally know and don't need to display? 😅

**Review by jhaals — COMMENTED** [review]
Hey everyone! I'm back from vacation, sorry that this has been laying around. Happy to merge this in, small question about the operating system but otherwise looking good :)

File: plugins/devtools-backend/src/api/DevToolsBackendApi.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (drodil) [reply plugins/devtools-backend/src/api/DevToolsBackendApi.ts:207]: Well, maybe not all devtools users know on which kind of instance the app is running. Don't see this hurting anyone :P

[the remark this answers is not in the document]
Comment (awanlin) [reply plugins/devtools-backend/src/api/DevToolsBackendApi.ts:207]: When I created this view I was very much inspired by the results from `backstage-cli info` which does include OS

[the remark this answers is not in the document]
Comment (adamdmharvey) [reply plugins/devtools-backend/src/api/DevToolsBackendApi.ts:207]: In a larger dev team, where some folks take care of different details, I can see some of the team not being aware of the exact OS.  Or even, "I know we use linux, but I'm not sure what flavor" if they've never touched the Dockerfile.  So yeah couldn't hurt, and in some cases could help for sure!

**Comment by awanlin** [comment]
Sweet! Thanks @drodil for this and your patience waiting for it to get merged 🚀 

Thanks fort he merge @jhaals 👍

**Pull request merged**